mmeval/metrics/matting_mse.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+from typing import Dict, List, Sequence
+
+from mmeval.core import BaseMetric
+
+
+def average(results, key):
+    """Average of key in results(list[dict]).
+
+    Args:
+        results (list[dict]): A list of dict containing the necessary data.
+        key (str): The key of target data.
+
+    Returns:
+        result: The average result.
+    """
+
+    total = 0
+    n = 0
+    for batch_result in results:
+        batch_size = batch_result.get('batch_size', 1)
+        total += batch_result[key] * batch_size
+        n += batch_size
+
+    return total / n
+
+
+class MattingMSE(BaseMetric):
+    """Mean Squared Error metric for image matting.
+
+    This metric compute per-pixel squared error average across all
+    pixels.
+    i.e. mean((a-b)^2) / norm_const
+
+    .. note::
+
+        Current implementation assume image / alpha / trimap array in numpy
+        format and with pixel value ranging from 0 to 255.
+
+    .. note::
+
+        pred_alpha should be masked by trimap before passing
+        into this metric
+
+    Default prefix: ''
+
+    Args:
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+        norm_const (int): Divide the result to reduce its magnitude.
+            Default to 1000.
+
+    Metrics:
+        - MattingMSE (float): Mean of Squared Error
+
+    Examples:
+
+        >>> from mmeval import MattingMSE
+        >>> import numpy as np
+        >>>
+        >>> mattingmse = MattingMSE()
+        >>> gts = np.random.randint(0, 255, size=(3, 32, 32))
+        >>> preds = np.random.randint(0, 255, size=(3, 32, 32))
+        >>> trimap = np.random.choice(a=(0,128,255), size=(3, 32, 32))
+        >>> mattingmse(preds, gts, trimap)  # doctest: +ELLIPSIS
+        {'MattingMSE': ...}
+    """
+
+    default_prefix = ''
+
+    def __init__(self, norm_const=1000, **kwargs) -> None:
+        self.norm_const = norm_const
+        super().__init__(**kwargs)
+
+    def add(self, pred_alphas: Sequence[np.ndarray], gt_alphas: Sequence[np.ndarray], trimaps: Sequence[np.ndarray]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add MattingMSE score of batch to ``self._results``
+
+        Args:
+        pred_alpha: Pred_alpha data of predictions.
+        ori_alpha: Ori_alpha data of data_batch.
+        ori_trimap: Ori_trimap data of data_batch.
```

Comment:
```suggestion
        """Add MattingMSE score of batch to ``self._results``

        Args:
            pred_alpha(Sequence): Pred_alpha data of predictions.
            ori_alpha(Sequence): Ori_alpha data of data_batch.
            ori_trimap(Sequence): Ori_trimap data of data_batch.
```

Reply:
This is my negligence. Now it has been revised in the latest submission. Thanks!